cli/cmd/diagnostics.go

```diff
@@ -0,0 +1,77 @@
+package cmd
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/linkerd/linkerd2/pkg/k8s"
+	"github.com/spf13/cobra"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	adminHTTPPortName string = "admin-http"
+)
+
+type diagnosticsOptions struct {
+	wait time.Duration
+}
+
+func newDiagnosticsOptions() *diagnosticsOptions {
+	return &diagnosticsOptions{
+		wait: 300 * time.Second,
+	}
+}
+
+func newCmdDiagnostics() *cobra.Command {
+	options := newDiagnosticsOptions()
+
+	cmd := &cobra.Command{
+		Use:   "diagnostics",
+		Short: "Fetch metrics directly from the Linkerd control plane containers",
+		Args:  cobra.NoArgs,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			k8sAPI, err := k8s.NewAPI(kubeconfigPath, kubeContext, impersonate, impersonateGroup, 0)
+			if err != nil {
+				return err
+			}
+
+			timeoutSeconds := int64(30)
+			deployments, err := k8sAPI.AppsV1().Deployments(controlPlaneNamespace).List(metav1.ListOptions{TimeoutSeconds: &timeoutSeconds})
+			if err != nil {
+				return err
+			}
+
+			// ensure we can connect to the public API before fetching the diagnostics.
+			checkPublicAPIClientOrRetryOrExit(time.Now().Add(options.wait), true)
```

Comment:
do we need this?

Reply:
umm, I am not really sure about this. 

I went through the architecture shown in the documentation [here](https://linkerd.io/2/reference/architecture/). Since the CLI talks to the Public-API, I felt that we should check if it can connect to the API or not. Maybe I am missing something?

Reply:
I think this subcommand gets everything it needs from port-forward and doesn't use the Linkerd public API at all.

Reply:
Understood. Thanks.